Tasks/BuildctlV0/src/utils.ts

```diff
@@ -0,0 +1,134 @@
+    } catch (error) {
+        tl.warning(tl.loc("BuildctlLatestNotKnown", buildctlLatestReleaseUrl, error, stableBuildctlVersion));
+    }
+
+    return stableBuildctlVersion;
+}
+export async function downloadBuildctl(version: string): Promise<string> {
+    var cachedToolpath = toolLib.findLocalTool(buildctlToolName, version);
+    let buildctlDownloadPath: string = null;
+    if (!cachedToolpath) {
+        try {
+            buildctlDownloadPath = await toolLib.downloadTool(getBuildctlDownloadURL(version), buildctlToolName + "-" + version + "-" + uuidV4() + getArchiveExtension());
+        } catch (exception) {
+            throw new Error(tl.loc("BuildctlDownloadFailed", getBuildctlDownloadURL(version), exception));
+        }
+
+        var unzipedBuildctlPath = await toolLib.extractTar(buildctlDownloadPath);
+        unzipedBuildctlPath = path.join(unzipedBuildctlPath, "bin", buildctlToolName);
+        tl.debug('Extracting archive: ' + unzipedBuildctlPath+' download path: '+buildctlDownloadPath);
+        var cachedToolpath = await toolLib.cacheFile(unzipedBuildctlPath, buildctlToolName, buildctlToolName, version);
+        tl.debug('CachedTool path: ' + cachedToolpath);    
+    }
+
+    var buildctlpath = findBuildctl(cachedToolpath);
+    if (!buildctlpath) {
+        throw new Error(tl.loc("BuildctlNotFoundInFolder", cachedToolpath))
+    }
+    tl.debug('Buildctl path: ' + buildctlpath);
+    fs.chmodSync(buildctlpath, "777");
+    return buildctlpath;
+}
+
+function getBuildctlDownloadURL(version: string): string {
+    switch (os.type()) {
+        case 'Windows_NT':
+            return util.format("https://github.com/moby/buildkit/releases/download/%s/buildkit-%s.windows-amd64.tar.gz", version, version);
+
+        case 'Darwin':
+            return util.format("https://github.com/moby/buildkit/releases/download/%s/buildkit-%s.darwin-amd64.tar.gz", version, version);
+
+        default:
+            case 'Linux':
+                return util.format("https://github.com/moby/buildkit/releases/download/%s/buildkit-%s.linux-amd64.tar.gz", version, version);
+
+    }
+}
+
+export async function getBuildKitPod(): Promise<string> {
+
+    var consistenthashkey = tl.getVariable('Build.Repository.Name')+tl.getInput("dockerFile", true);
+    var kubectlToolPath = tl.which("kubectl", true);
+    var kubectlTool = tl.tool(kubectlToolPath);
+    kubectlTool.arg('get');
+    kubectlTool.arg('services');
+    kubectlTool.arg('k8s-poolprovider');
+    kubectlTool.arg('-n=azuredevops');
+    kubectlTool.arg('-o=json');
+    var serviceResponse= kubectlTool.execSync();
+        
+    //console.log("PodName: "+ JSON.parse(serviceResponse.stdout).status.loadBalancer.ingress[0].ip);
+    var clusteruri = "http://"+JSON.parse(serviceResponse.stdout).status.loadBalancer.ingress[0].ip+":8082/consistenthash";
+    let request = new webclient.WebRequest();
+        
+    request.uri = clusteruri;
+    let headers = {
+        "key": consistenthashkey
```

Comment:
please don't use the word consistentHash anywhere in the task code.

The responsibility of the task is to ask the service to return a valid buildkit pod. How the service decides to do it - round robin, consistent hash or calling the DRI to get the pod name is totally its responsibility.